app/controllers/collections_controller.rb

```diff
@@ -0,0 +1,13 @@
+class CollectionsController < ApplicationController
+  def index
+    if (@user = User.find_by(username: params[:username]))
+      @collections = Collection.where(user_id: @user.id)
+    elsif (@user = Organization.find_by(slug: params[:username]))
+      @collections = Collection.where(organization_id: @user.id)
+    end
+  end
+
+  def show
+    @collection = Collection.includes(%i[user articles]).find_by(path: "/#{params[:username].downcase}/series/#{params[:slug]}")
```

Comment:
I don't think we need to store the path, as storing the path would mean we'd need to keep it updated in case the user changes username or if the slug changes somehow and so on, in addition to having to populate the field for existing collections.

Though it would result in one more query I don't think that's a huge deal:

- load the user through the username: `user = User.find_by!()`
- load the collection through `user.collections.find_by!()`

Reply:
This in general sounds like a good way to go, but we have the problem of the slug not being a real slug again. Any clever idea on how to proceed? I see there's a `titleize` method but it doesn't quite work for our case.

Reply:
For now I'm using the slug as is, although this produces ugly URLs, so it should be changed.